samples/matter/thermostat/src/app_event.h

```diff
@@ -8,38 +8,19 @@
 
 #include <cstdint>
 
-#include "event_types.h"
+enum class AppEventType : uint8_t { None = 0, TemperatureButton };
 
-class LEDWidget;
-
-enum class AppEventType : uint8_t {
-	None = 0,
-	Button,
-	ButtonPushed,
-	ButtonReleased,
-	Timer,
-	UpdateLedState,
-	IdentifyStart,
-	IdentifyStop,
-	Thermostat
-};
-
-enum class FunctionEvent : uint8_t { NoneSelected = 0, FactoryReset, SoftwareUpdate, AdvertisingStart };
+enum class TemperatureButtonAction : uint8_t { Pushed, Released };
 
 struct AppEvent {
```

Comment:
imho, these `AppEvent` structures can probably be removed in most cases and arguments like `TemperatureButtonAction` can be captured by a lambda directly to reduce the code size even further, but this can be done in a followup PR :)

Reply:
That is a good idea, it simplifies code a lot :)